### PR TITLE
Testfixes

### PR DESCRIPTION
**Pull request by Vampire**
The changes from this PR were necessary to be able to build hazelcast on Windows. Should work on Windows and Linux now with these changes.

### CODE REVIEW

**Comment by Vampire** [comment]
Hey there, is anything wrong with this PR? :-)

**Comment by bilalyasar** [comment]
verify

**Comment by bilalyasar** [comment]
run-lab-run

**Comment by jerrinot** [comment]
@Vampire: Many thanks for the contribution. 
The explanation is simpler: We simple forgot about it. Let me have a look.

**Comment by Vampire** [comment]
Hm, what did fail? I cannot look at the check result.

**Comment by Donnerbart** [comment]
`com.hazelcast.mapreduce.MapReduceTest.testMapperReducerChunked`

**Error Message**
test timed out after 60000 milliseconds

**Stacktrace**
```
org.junit.runners.model.TestTimedOutException: test timed out after 60000 milliseconds
	at java.lang.Object.wait(Native Method)
	at java.lang.Object.wait(Object.java:485)
	at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.get(InvocationFuture.java:187)
	at com.hazelcast.map.impl.proxy.MapProxySupport.invokeOperation(MapProxySupport.java:321)
	at com.hazelcast.map.impl.proxy.MapProxySupport.putInternal(MapProxySupport.java:292)
	at com.hazelcast.map.impl.proxy.MapProxyImpl.put(MapProxyImpl.java:105)
	at com.hazelcast.map.impl.proxy.MapProxyImpl.put(MapProxyImpl.java:95)
	at com.hazelcast.mapreduce.MapReduceTest.testMapperReducerChunked(MapReduceTest.java:490)
```

**Comment by Donnerbart** [comment]
run-lab-run

**Comment by Vampire** [comment]
That link would also not have helped, no access. But ok, if it is unrelated, I'm satisfied :-)

**Comment by bilalyasar** [comment]
com.hazelcast.spi.discovery.DiscoverySpiTest.test_AbstractDiscoveryStrategy_getOrNull
```
java.lang.NoSuchFieldException: theCaseInsensitiveEnvironment
	at java.lang.Class.getDeclaredField(Class.java:1884)
	at com.hazelcast.spi.discovery.DiscoverySpiTest.setEnvironment(DiscoverySpiTest.java:256)
	at com.hazelcast.spi.discovery.DiscoverySpiTest.test_AbstractDiscoveryStrategy_getOrNull(DiscoverySpiTest.java:211)
```

File: hazelcast/src/test/java/com/hazelcast/spi/discovery/DiscoverySpiTest.java [hunk 1]
```diff
@@ -252,6 +252,11 @@ private static void setEnvironment(String key, String value) throws Exception {
                 map.put(key, value);
             }
         }
+
+        Field field = Class.forName("java.lang.ProcessEnvironment").getDeclaredField("theCaseInsensitiveEnvironment");
```
Comment (jerrinot) [inline hazelcast/src/test/java/com/hazelcast/spi/discovery/DiscoverySpiTest.java]: can you please explain reasoning behind this? 
this whole method is a mystery to me and this adds additional sorcery into the mix:)

Comment (Vampire) [inline hazelcast/src/test/java/com/hazelcast/spi/discovery/DiscoverySpiTest.java]: Like the part above it, it adds to faking the set environment variables.
Some methods use this case insensitive environment, some the other that is manipulated.
Maybe this is different between Windows and *nix Java versions, though I thought I tested on both.
So maybe I should add some failsafety like in the block above it.

Comment (mesutcelik) [inline hazelcast/src/test/java/com/hazelcast/spi/discovery/DiscoverySpiTest.java]: Hi @noctarius ,
What is the whole purpose of `setEnviroment` method? Can you please shed some light? cc: @kobalski

Comment (noctarius) [inline hazelcast/src/test/java/com/hazelcast/spi/discovery/DiscoverySpiTest.java]: Sorry for the late answer. It adds system environment properties at runtime which normally would have been set on command line.

**Comment by mesutcelik** [comment]
run-lab-run

**Comment by mesutcelik** [comment]
The same error... It could that is not working with the JDK version of PR Builder.

any idea @Vampire ?

```
java.lang.NoSuchFieldException: theCaseInsensitiveEnvironment
	at java.lang.Class.getDeclaredField(Class.java:1884)
	at com.hazelcast.spi.discovery.DiscoverySpiTest.setEnvironment(DiscoverySpiTest.java:286)
	at com.hazelcast.spi.discovery.DiscoverySpiTest.test_AbstractDiscoveryStrategy_getOrNull(DiscoverySpiTest.java:241)
```

**Comment by jerrinot** [comment]
@Vampire: can you please have a look at the failure?
```
java.lang.NoSuchFieldException: theCaseInsensitiveEnvironment
    at java.lang.Class.getDeclaredField(Class.java:1884)
    at com.hazelcast.spi.discovery.DiscoverySpiTest.setEnvironment(DiscoverySpiTest.java:286)
    at com.hazelcast.spi.discovery.DiscoverySpiTest.test_AbstractDiscoveryStrategy_getOrNull(DiscoverySpiTest.java:241)
```

**Comment by jerrinot** [comment]
I am closing this due inactivity. @Vampire feel free to re-open it when you have more time.

**Pull request closed without merging**